tests/__setup.ts

```diff
@@ -0,0 +1,66 @@
+import ts from 'typescript'
+;(global as any).TS_FORMAT_PROJECT_ROOT = 'src/'
+import * as tsm from 'ts-morph'
+
+const formatTSDiagonsticsForJest = (
+  diagnostics: readonly tsm.Diagnostic[],
+): string => {
+  const tsDiagnostics = diagnostics.map(d => d.compilerObject)
+  const formatHost: ts.FormatDiagnosticsHost = {
+    getCanonicalFileName: path => path,
+    getCurrentDirectory: ts.sys.getCurrentDirectory,
+    getNewLine: () => ts.sys.newLine,
+  }
+
+  const tsReport = ts.formatDiagnosticsWithColorAndContext(
+    tsDiagnostics as any,
+    formatHost,
+  )
+
+  const sourcePath =
+    process.cwd() +
+    '/' +
+    (((global as any).TS_FORMAT_PROJECT_ROOT as string) || '')
+
+  const summaryReport = `${
+    tsDiagnostics.length
+  } Type Error(s):\n\n${tsDiagnostics
+    .map(d =>
+      d.file ? d.file.fileName.replace(sourcePath, '') : '<unknown file>',
+    )
+    .join('\n')}`
+
+  const jestReport = `${summaryReport}\n\n${tsReport}`
+
+  return jestReport
+}
+
+expect.extend({
+  toTypeCheck(tsConfigFilePath: string) {
```

Comment:
I built similar [for nexus](https://github.com/prisma/nexus/blob/develop/tests/_setup.ts). Inline for now but I could see/I think we should extract this out into a jest-ts-test-utils package.